### ISSUE TITLE
Remove need of `png.h` in INCLUDE_PATH

### ISSUE THREAD

**Issue by ax3l** [pngwriter/pngwriter#57]
Since `pngwriter.h` is the general include that users include, the [third party includes](https://github.com/pngwriter/pngwriter/blob/c88a1cac4217eabfca6990ed005702533d1af39b/src/pngwriter.h#L55-L68) of `png.h`, `zlib.h` and `ft2build.h` are required to reside in their `INCLUDE_PATH`.

That's not necessary (since we don't use them in interfaces) and a pain with non-default paths.

Related to:
 - https://github.com/ComputationalRadiationPhysics/cmake-modules/issues/4
 - #7

Found by @psychocoderHPC with a non-default `libpng`.

**Comment by ax3l** [comment]
Due to public member functions such as
```c++
#ifndef NO_FREETYPE
   void my_draw_bitmap( FT_Bitmap * bitmap, int x, int y, double red, double green, double blue);
   void my_draw_bitmap_blend( FT_Bitmap * bitmap, int x, int y,double opacity,  double red, double green, double blue);
#endif
```

that pull in externally exposed `FT_Bitmap` types from `<ft2build.h>` which in turn *must* be included after `<png.h>` might prevent implementing #7 or the whole issue without an interface change.

**Comment by individual61** [comment]
Yeah, I came here to verify that they are public, because that didn't sound right!

**Comment by ax3l** [comment]
hehe, that was fast :)

**Comment by ax3l** [comment]
argh, there are more interfaces such as
```c++
    int read_png_info(FILE *fp, png_structp *png_ptr, png_infop *info_ptr);
    int read_png_image(FILE *fp, png_structp png_ptr, png_infop info_ptr,
```

**Comment by ax3l** [comment]
commit in a stalled test branch: https://github.com/ax3l/pngwriter/tree/topic-hideThirdPartyIncludes
needs more refactoring in upcoming releases
